### PR TITLE
Added papers on AAAI2019 and NIPS2018

### PR DESCRIPTION
**Pull request by alphadl**
Added three papers about Machine Translation on AAAI 2019, all of them are from Tecent AI lab.

[1] Baosong Yang, Jian Li, Derek Wong, Lidia S. Chao, Xing Wang, and Zhaopeng Tu. ***Context-Aware Self-Attention Networks***. AAAI 2019. --> personally I divided it into class **NMT_robustness** because it's neither novel framework nor document translation

[2] Zi-Yi Dou, Zhaopeng Tu, Xing Wang, Longyue Wang, Shuming Shi, and Tong Zhang. ***Dynamic Layer Aggregation for Neural Machine Translation with Routing-by-Agreement***. AAAI 2019. --> **NMT_model_architecture**

[3] Xiang Kong, Zhaopeng Tu, Shuming Shi, Eduard Hovy, and Tong Zhang. ***Neural Machine Translation with Adequacy-Oriented Learning***. AAAI 2019. --> **NMT_training_objectives_and_frameworks**

( Please feel free to correct me if you feel there is any inappropriate classification

### CODE REVIEW

**Comment by minicheshire** [comment]
Thanks a lot! As for the categorization issue, my opinion is listed below:

- ***Context-Aware Self-Attention Networks***.  I would recommend to categorize it in the section of **Attention Mechanism**. This paper proposed to capture the global context information and make use of the representations of a word from the lower layers in a manner of attention mechanism. Personally speaking, it reminds me of the paper from  COLING 2018 - ***Multi-layer Representation Fusion for Neural Machine Translation***, which proposed to improve the internal representation with the information from lower layers. I'd like to say, however, that the COLING'18 work has implemented the idea with two different approaches - FNN and multi-hop self-attention (see Section 3.2). Since it is not limited to the attention mechanism, I chose to categorize the paper in the section of *Model Architecture*. (They didn't try fusing the context information as this AAAI'19 work does.)
In fact, this "attention between layers" idea is also seen in ***Training Deeper Neural Machine Translation Models with Transparent Attention*** from EMNLP 2018, which proposed to fully connect all the encoder layers and the encoder-decoder attention modules with a weight vector and update the weight with softmax. While the motivations of it and the AAAI'19 work might be different, in my eyes, they still somewhat share similarities in methodology. 

- ***Dynamic Layer Aggregation for Neural Machine Translation with Routing-by-Agreement***. I would recommend to categorize it in the section of **Model Architecture**. In my opinion, the highlight of this paper is the proposal of a capsule-like architecture for NMT. The dynamic routing mechanism is experimented as well, which is beyond attention mechanism.

- ***Neural Machine Translation with Adequacy-Oriented Learning***. As for this paper, I would say it's about **Coverage Constraints**. As you can see in Fig.2, there's a coverage module, which is further oriented by the proposed orientator. The focus of the coverage mechanism is to improve adequacy of NMT, which is a kind of prior knowledge. As a result, I'd suggest it in the category of **Coverage Constraints**.

Before I merge your pr, do let me know if you have different opinions of the categorization!

**Comment by alphadl** [comment]
Thx for @minicheshire , Your analysis is very thorough, I resubmitted the RP, where the new version is classified according to what you said.

**Comment by minicheshire** [comment]
Thanks! XD
But it seems that your last commit accidentally dropped out the ***Neural Machine Translation with Adequacy-Oriented Learning*** _(:ᗤ」ㄥ)_
Besides, the ***Context-Aware Self-Attention Networks*** belongs to the **Attention Mechanism** section; and in README.md please change the category of the ***Dynamic Layer Aggregation for Neural Machine Translation with Routing-by-Agreement*** paper as well~

**Comment by minicheshire** [comment]
Thanks! It seems that the ***Neural Machine Translation with Adequacy-Oriented Learning*** paper is still missing... But never mind, I'll fix that later.
Thanks again for your contributions!

**Comment by minicheshire** [comment]
Ah... This time I did't tell it right. You've made it perfect!
